worldedit-core/src/main/java/com/fastasyncworldedit/core/queue/implementation/blocks/IntBlocks.java

```diff
@@ -0,0 +1,268 @@
+package com.fastasyncworldedit.core.queue.implementation.blocks;
+
+import com.fastasyncworldedit.core.Fawe;
+import com.fastasyncworldedit.core.queue.IBlocks;
+import com.fastasyncworldedit.core.queue.IChunkSet;
+import com.sk89q.worldedit.internal.util.LogManagerCompat;
+import com.sk89q.worldedit.world.block.BlockState;
+import com.sk89q.worldedit.world.block.BlockTypesCache;
+import org.apache.logging.log4j.Logger;
+
+import javax.annotation.Nullable;
+
+public abstract class IntBlocks implements IBlocks {
+
+    private static final Logger LOGGER = LogManagerCompat.getLogger();
+
+    protected static final IntBlocks.Section FULL = new IntBlocks.Section() {
+        @Override
+        public int[] get(IntBlocks blocks, int layer) {
+            return blocks.blocks[layer];
+        }
+
+        // Ignore aggressive switch here.
+        @Override
+        public int[] get(IntBlocks blocks, int layer, boolean aggressive) {
+            return blocks.blocks[layer];
+        }
+
+        @Override
+        public boolean isFull() {
+            return true;
+        }
+    };
+    protected static final IntBlocks.Section EMPTY = new IntBlocks.Section() {
+        @Override
+        public int[] get(IntBlocks blocks, int layer) {
+            // Defaults to aggressive as it should only be avoided where we know we've reset a chunk during an edit
+            return get(blocks, layer, true);
+        }
+
+        @Override
+        public int[] get(IntBlocks blocks, int layer, boolean aggressive) {
+            synchronized (blocks.sectionLocks[layer]) {
+                if (blocks.sections[layer] == FULL) {
+                    return FULL.get(blocks, layer);
+                }
+                int[] arr = blocks.blocks[layer];
+                if (arr == null) {
```

Comment:
We could introduce a package private method helper class to reuse common parts across several classes in the `blocks` package.

Reply:
In regards to a helper class, I'll likely wait and do it in a future PR as suggested by @SirYwell. Don't want to clutter this PR up with too many different things.